source/Cosmos.System.Plugs/System/IO/DirectoryInfoImpl.cs

```diff
@@ -36,13 +36,19 @@ public static class DirectoryInfoImpl
             Global.mFileSystemDebugger.SendInternal(aPath);
 
             aStorage = VFSManager.GetDirectory(aPath);
+
+            if (aStorage == null)

```

Comment:
Don't throw an exception here.

MSDN remarks:
This constructor does not check if a directory exists. This constructor is a placeholder for a string that is used to access the disk in subsequent operations.